pkg/types/unix_milli_test.go

```diff
@@ -28,3 +28,69 @@ func TestUnixMilli_MarshalJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestUnixMilli_UnmarshalJSON(t *testing.T) {
+	subtests := []struct {
+		name   string
+		input  UnixMilli
+		output string
+	}{
+		{"zero", UnixMilli{}, "null"},
+		{"epoch", UnixMilli(time.Unix(0, 0)), "0"},
+		{"nonzero", UnixMilli(time.Unix(1234567890, 62000000)), "1234567890062"},
+	}
+
+	for _, st := range subtests {
+		t.Run(st.name, func(t *testing.T) {
+			var s UnixMilli
+			err := s.UnmarshalJSON([]byte(st.output))
+
+			require.NoError(t, err)
+			require.Equal(t, st.input, s)
+		})
+	}
+}
+
+func TestUnixMilli_MarshalText(t *testing.T) {
+	subtests := []struct {
+		name   string
+		input  UnixMilli
+		output string
```

Comment:
The input/output variables are mixed up here. 

Also, the pretty much same test data is repeated multiple times here. Maybe create a global variable for it that's something like []struct{name, time, json, text}` (if JSON and text representation even differ anywhere, otherwise a common field for both).